xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -663,6 +679,13 @@ posix_mknod(call_frame_t *frame, xlator_t *this, loc_t *loc, mode_t mode,
     posix_set_parent_ctime(frame, this, par_path, -1, loc->parent, &postparent);
 
     op_ret = 0;
+    if (atomic_creation) {
+        op_ret = sys_rename(real_path, orig_real_path);
+        if (op_ret)
+            gf_log(this->name, GF_LOG_ERROR,
```

Comment:
On failure we also need to cleanup all the created entries.

Reply:
That is already happening if op_ret is 0 we are calling unlink(real_path)